app/src/test/java/com/google/samples/apps/sunflower/data/PlantTest.kt

```diff
@@ -39,10 +40,37 @@ class PlantTest {
     }
 
     @Test fun test_shouldBeWatered() {
-        assertFalse(plant.shouldBeWatered(Calendar.getInstance()))
-        assertFalse(plant.shouldBeWatered(Calendar.getInstance().apply { add(DAY_OF_YEAR, -1) }))
-        assertFalse(plant.shouldBeWatered(Calendar.getInstance().apply { add(DAY_OF_YEAR, -2) }))
-        assertTrue(plant.shouldBeWatered(Calendar.getInstance().apply { add(DAY_OF_YEAR, -3) }))
+        Calendar.getInstance().let { now ->
+            // Generate lastWateringDate from being as copy of now.
+            val lastWateringDate = Calendar.getInstance()
+
+            // Test for lastWateringDate is today.
+            lastWateringDate.time = now.time
+            assertFalse(plant.shouldBeWatered(now, lastWateringDate.apply { add(DAY_OF_YEAR, -0) }))
+
+            // Test for lastWateringDate is yesterday.
+            lastWateringDate.time = now.time
+            assertFalse(plant.shouldBeWatered(now, lastWateringDate.apply { add(DAY_OF_YEAR, -1) }))
+
+            // Test for lastWateringDate is the day before yesterday.
+            lastWateringDate.time = now.time
+            assertFalse(plant.shouldBeWatered(now, lastWateringDate.apply { add(DAY_OF_YEAR, -2) }))
+
+            // Test for lastWateringDate is some days ago, three days ago, four days ago etc.
+            lastWateringDate.time = now.time
+            val days = now.getMoreThanTwoDaysFromNow()
+            assertTrue(plant.shouldBeWatered(now, lastWateringDate.apply { add(DAY_OF_YEAR, -days) }))
+        }
+    }
+
+    private fun Calendar.getMoreThanTwoDaysFromNow(): Int {
```

Comment:
I think that using `Random()` here can cause unpredictable test failures should the details of `plant` change.  Prefer to replace `-days` in line 62 with `-3`.